File: runescape.gamepack/src/main/java/io/runebox/ObfInfo.java
package io.runebox;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.ANNOTATION_TYPE, ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.TYPE_USE, ElementType.METHOD, ElementType.FIELD, ElementType.LOCAL_VARIABLE, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
public @interface ObfInfo {
	String owner() default "";

	String name() default "";

	String desc() default "";

	String opaque() default "";

	int intMultiplier() default 0;

	long longMultiplier() default 0L;
}

File: internal/runebox.updater/src/main/kotlin/io/runebox/updater/asm/ObfInfo.kt
package io.runebox.updater.asm

import org.objectweb.asm.tree.ClassNode

data class ObfInfo(
    val owner: String,
    val name: String,
    val desc: String,
    val opaque: String,
    val intMultiplier: Int,
    val longMultiplier: Long,
)

val ClassNode.obf: ObfInfo? get() {
    val anno = (this.visibleAnnotations ?: emptyList()).find("ObfInfo") ?: return null
    return ObfInfo(
        anno["owner", ""],
        anno["name", ""],
        anno["desc", ""],
        anno["opaque", ""],
        anno["intMultiplier", 0],
        anno["longMultiplier", 0L]
    )
}